tests/unit/s2n_random_test.c

```diff
@@ -142,6 +146,110 @@ static S2N_RESULT s2n_basic_pattern_tests(S2N_RESULT (*s2n_get_random_data_cb)(s
     return S2N_RESULT_OK;
 }
 
+static void swap(size_t i, size_t j,
+    uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS]) {
+    uint64_t temp = range_results[j];
+    range_results[j] = range_results[i];
+    range_results[i] = temp;
+}
+
+static void sort_array(uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS]) {
+    for (size_t i = 1; i < NUMBER_OF_RANGE_FUNCTION_CALLS; i++) {
+        for (size_t j = i; j > 0; j--) {
+            if (range_results[j - 1] > range_results[j]) {
+                swap(j - 1, j, range_results);
+            }
+            else {
+                /* Invariant: subarray [0, i-1] is sorted */
+                continue;
+            }
+        }
+    }
+}
+
+static S2N_RESULT s2n_tests_get_range(S2N_RESULT (*s2n_get_range_cb)(int64_t bound, uint64_t *output))
```

Comment:
Why do we need to include a callback in the arguments here? We're only testing s2n_public_random.